src/pyodide/internal/python.js

```diff
@@ -278,32 +335,113 @@ const SNAPSHOT_IMPORTS = [
 ];
 
 /**
- * Create memory snapshot by importing SNAPSHOT_IMPORTS to ensure these packages
- * are initialized in the linear memory snapshot and then saving a copy of the
- * linear memory into MEMORY.
+ * Python modules do a lot of work the first time they are imported. The memory
+ * snapshot will save more time the more of this work is included. However, we
+ * can't snapshot the JS runtime state so we have no ffi. Thus some imports from
+ * user code will fail.
+ *
+ * If we are doing a baseline snapshot, just import everything from
+ * SNAPSHOT_IMPORTS. These will all succeed.
+ *
+ * If doing a script-specific "dedicated" snap shot, also try to import each
+ * user import.
+ *
+ * All of this is being done in the __main__ global scope, so be careful not to
+ * pollute it with extra included-by-default names (user code is executed in its
+ * own separate module scope though so it's not _that_ important).
  */
-async function makeLinearMemorySnapshot(emscriptenModule) {
-  // TODO: make memory snapshot with more than just these hardcoded imports, use real script imports.
+function memorySnapshotDoImports(Module) {
```

Comment:
My feelings here are:
* `emscriptenModule` is more explicit
* it's a bit long for something used so much

The main reason I want to change it is that everywhere in the emscripten and pyodide code bases it's called `Module` so I spend a significant amount of time debugging copy paste errors because it takes ~30 seconds to build and run, another 30 seconds to understand the traceback in good cases and in bad cases much more. If our tests ran faster, our logs were easier to understand, or we had static analysis tools that worked on our javascript codebase I would be less eager to keep the name consistent. As it is, each extra little paper cut causes a lot of increased development time.

OTOH, I am willing to change it back if you feel strongly about it.